File: agcapi/AGCAPIDemo/UserInfoViewController.m
//
// Copyright 2020. Huawei Technologies Co., Ltd. All rights reserved.
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
//

#import "UserInfoViewController.h"
#import <AGConnectAuth/AGConnectAuth.h>

@interface UserInfoViewController ()

@property (weak, nonatomic) IBOutlet UILabel *uid;
@property (weak, nonatomic) IBOutlet UILabel *name;

@end

@implementation UserInfoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"User Info";
    
    self.uid.text = [AGCAuth getInstance].currentUser.uid;
    self.name.text = [AGCAuth getInstance].currentUser.displayName;
}

- (IBAction)logout:(id)sender {
    [[AGCAuth getInstance] signOut];
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: AGCAPI-objectiveC/AGCAPIDemo-Swift/UserInfoViewController.swift
//
//  Copyright (c) Huawei Technologies Co., Ltd. 2020. All rights reserved
//

import UIKit
import AGConnectAuth

class UserInfoViewController: UIViewController {

    @IBOutlet weak var uid: UILabel!
    @IBOutlet weak var name: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "User Info"
        
        uid.text = AGCAuth.instance().currentUser?.uid
        name.text = AGCAuth.instance().currentUser?.displayName
    }
    
    @IBAction func logout(_ sender: Any) {
        AGCAuth.instance().signOut()
        self.navigationController?.popViewController(animated: true)
    }
}
